src/verify/parse.ts

```diff
@@ -41,6 +42,57 @@ export function getTypescriptVersionRequirement(text: string): string | undefine
     return undefined;
 }
 
+export function getNormalizedTypescriptVersion(): string {
+    const tsVersion = new semver.SemVer(ts.version);
+    // remove prerelease suffix when matching to allow testing with nightly builds
+    return `${tsVersion.major}.${tsVersion.minor}.${tsVersion.patch}`;
+}
+
+export function preprocessDirectives(text: string): string {
+    if (!/^#(?:if|else|endif)\b/m.test(text)) {
+        return text; // If there are no directives, just return the input unchanged
+    }
+    const tsVersion = getNormalizedTypescriptVersion();
+    const enum State {
+        Initial,
+        If,
+        Else,
+    }
+    const lines = text.split(/\n/);
+    const result = [];
+    let collecting = true;
+    let state = State.Initial;
+    for (const line of lines) {
+        if (line.startsWith("#if typescript")) {
+            if (state !== State.Initial) {
+                throw lintSyntaxError("#if directives cannot be nested");
+            }
+            state = State.If;
+            collecting = semver.satisfies(tsVersion, line.slice(14).trim());
```

Comment:
instead of `.slice(14)`, can you include the string literal you're splitting with? `"if typescript ".length`